Source/WebCore/Modules/cache/WorkerCacheStorageConnection.cpp

```diff
@@ -91,24 +91,14 @@ WorkerCacheStorageConnection::~WorkerCacheStorageConnection()
     callOnMainThread([mainThreadConnection = WTFMove(m_mainThreadConnection)]() mutable { });
 }
 
-void WorkerCacheStorageConnection::open(const ClientOrigin& origin, const String& cacheName, CacheIdentifierCallback&& callback)
+auto WorkerCacheStorageConnection::open(const ClientOrigin& origin, const String& cacheName) -> Ref<OpenPromise>
 {
-    uint64_t requestIdentifier = ++m_lastRequestIdentifier;
-    m_openCachePendingRequests.add(requestIdentifier, WTFMove(callback));
-
-    callOnMainThread([workerThread = Ref { m_scope.thread() }, mainThreadConnection = m_mainThreadConnection, requestIdentifier, origin = origin.isolatedCopy(), cacheName = cacheName.isolatedCopy()] () mutable {
-        mainThreadConnection->open(origin, cacheName, [workerThread = WTFMove(workerThread), requestIdentifier] (const CacheIdentifierOrError& result) mutable {
-            workerThread->runLoop().postTaskForMode([requestIdentifier, result] (auto& scope) mutable {
-                downcast<WorkerGlobalScope>(scope).cacheStorageConnection().openCompleted(requestIdentifier, result);
-            }, WorkerRunLoop::defaultMode());
-        });
+    OpenPromise::Producer producer;
```

Comment:
return invokeAsync(RunLoop::mainthread(), []... { Return mainThreadCinnection->open(...)

No need for a producer